org.jacoco.core.test/src-kotlin/org/jacoco/core/test/validation/KotlinGeneratedTest.java

```diff
@@ -0,0 +1,37 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2018 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Fabian Mastenbroek - initial API and implementation
+ *    
+ *******************************************************************************/
+package org.jacoco.core.test.validation;
+
+import static org.junit.Assert.assertNotNull;
+
+import org.jacoco.core.analysis.IClassCoverage;
+import org.jacoco.core.analysis.ICounter;
+import org.jacoco.core.test.validation.targets.KotlinDataClassTarget;
+import org.junit.Test;
+
+/**
+ * Tests of generated methods for a data class.
+ */
+public class KotlinGeneratedTest extends ValidationTestBase {
+
+	public KotlinGeneratedTest() {
+		super("src-kotlin", KotlinDataClassTarget.class);
+	}
+
+	@Test
+	public void testCovered() {
+		final IClassCoverage classCoverage =
+				coverageForClass(KotlinDataClassTarget.DataClass.class);
+		assertNotNull(classCoverage);
+		assertStatus(classCoverage, ICounter.FULLY_COVERED);
```

Comment:
IMO we also need negative test, i.e. when it is not covered, otherwise we won't catch case when filter excludes too much. See https://github.com/jacoco/jacoco/blob/ce7de98c76b8bd107c05a061193de76d21c4c205/org.jacoco.core.test/src/org/jacoco/core/test/validation/ControlStructuresTest.java#L33-L41